packages/react-server/src/ReactFlightServer.js

```diff
@@ -200,10 +200,21 @@ function attemptResolveElement(
       return [REACT_ELEMENT_TYPE, type, key, props];
     }
     switch (type.$$typeof) {
+      case REACT_LAZY_TYPE: {
+        const payload = type._payload;
+        const init = type._init;
+        const wrappedType = init(payload);
+        return attemptResolveElement(wrappedType, key, ref, props);
+      }
       case REACT_FORWARD_REF_TYPE: {
         const render = type.render;
         return render(props, undefined);
       }
+      case REACT_ELEMENT_TYPE: {
+        // this can happen when a lazy component resolves to an element instead of
+        // a Component.
+        return attemptResolveElement(type.type, type.key, type.ref, type.props);
+      }
```

Comment:
I think the solution is to just remove this and then let it fall through to the error case. You could also put custom error messages at the bottom of this case if you want to explain in plain words to the user what's wrong.

Semantically this is not right since it's dropping key, ref and props from the parent.

Reply:
yup, the need for this went away when I added lazy resolution to `resolveModelToJSON`